galter_subjects_utils/contrib/lcsh/converter.py

```diff
@@ -115,25 +135,47 @@ def find_explanation(nodes_of_change_note):
     return ""
 
 
+pattern = r"covered by .+ subject heading( {(?P<subject_new>.+)})? .+(?P<id_new>sh\s?\d+)$"  # noqa
+regex_for_replacement = re.compile(pattern)
+
+
+def extract_replacement_id_subject(notes):
+    """Returns (id_new, subject_new) tuple from notes or ("", "") else."""
+    matches = regex_for_replacement.search(notes)
+    if not matches:
+        return "", ""
+    id_new = matches.group("id_new") or ""
+    id_new = id_new.replace(" ", "")  # get rid of potential space after 'sh'
+    lcsh = LCSHScheme()
+    id_new = lcsh.generate_id(id_new) if id_new else id_new
+    subject_new = matches.group("subject_new") or ""
+    return id_new, subject_new
+
+
 def to_raw_replacement(topic):
     """Fill out replacement dict."""
     node_main = find_main_node(topic)
     node_deprecation = find_deprecation_node(topic, node_main)
+    notes = find_explanation(node_main.get("skos:changeNote"))
+    id_new, subject_new = extract_replacement_id_subject(notes)
     return {
         "id": node_main.get("@id").replace("http://", "https://"),
         "time": node_deprecation.get("cs:createdDate", {}).get("@value"),
         "subject": node_main.get("skosxl:literalForm", {}).get("@value"),
-        "new_id": "TODO:Expert:Fill me",
-        "new_subject": "TODO:Expert:Fill me",
-        "notes": find_explanation(node_main.get("skos:changeNote")),
+        "new_id": id_new,
+        "new_subject": subject_new,
+        "notes": notes,
```

Comment:
Yeah... it was hard for me to find an order that made most sense. I also considered the following:

```
{
    "id": ...,
    "subject": ...,
    "new_id": ...,
    "new_subject": ...,
    "time": ...,
    "notes": ...
}
```

Since the pair `(id, subject)` (resp. `(new_id, new_subject)`) goes together as one entity and that reflects it better. 

In the end, I went with that order because it's the same as the one laid out in the CSV file, where the primary key is the `id` and the primary filtering is the `time`. I will keep this in mind and see if what works best for Gretchen.